torch/testing/_internal/common_utils.py

```diff
@@ -1866,7 +1866,8 @@ def to_dense(tensor):
             def partial_to_dense(tensor):
                 if tensor.layout not in compressed_sparse_layouts or tensor.values().ndim == 1:
                     return tensor.to_dense()
-                return torch.stack([partial_to_dense(sub_tensor) for sub_tensor in tensor])
+                lst = [partial_to_dense(torch.select_copy(tensor, 0, i)) for i in range(len(tensor))]
+                return torch.stack(lst) if lst else tensor.to_dense()
```

Comment:
What about a follow-up that implements `narrow`? Then we can use it here.

Reply:
No. The code here is a temporary hack to circumvent the issue of `to_dense` not being implemented for hybrid sparse compressed tensors. The corresponding TODO task is described here:
https://github.com/pytorch/pytorch/blob/55e55d95ea9a6f64bba50cdc9e243808cb534202/torch/testing/_internal/common_utils.py#L1849-L1852